app/models/charge.js

```diff
@@ -89,26 +89,22 @@ module.exports = correlationId => {
     return response.body
   }
 
-  const patch = function (chargeId, op, path, value, subSegment) {
-    return new Promise(function (resolve, reject) {
+  const patch = async function (chargeId, op, path, value, subSegment) {
+    try {
       const payload = {
         op: op,
         path: path,
         value: value
       }
-      connectorClient({ correlationId }).patch({ chargeId, payload }, subSegment)
-        .then(response => {
-          const code = response.statusCode
-          if (code === 200) {
-            resolve()
-          } else {
-            reject(new Error('Calling connector to patch a charge returned an unexpected status code'))
-          }
-        })
-        .catch(err => {
-          reject(err)
-        })
-    })
+      const response = await connectorClient({ correlationId }).patch({ chargeId, payload }, subSegment)
+      if (response.statusCode === 200) {
+        return
+      } else {
+        throw new Error('Calling connector to patch a charge returned an unexpected status code')
+      }
+    } catch (err) {
+      throw err
```

Comment:
does catching and then throwing do something different to doing nothing?

Reply:
i'm not sure. the code used to be
```.catch(err => {	
          reject(err)	
        })
```
so I assumed throwing the `err` was enough. I'll throw a new Error as it seems that's the pattern though.

Reply:
I reckon it had to do this because the code was manually returning a promise, but now we're using async/await which automatically wraps the function in a promise so any errors thrown will cause the promise to be rejected.